src/lib/modules/blockchain_process/proxy.js

```diff
@@ -119,77 +135,98 @@ exports.serve = async (ipc, host, port, ws, origin) => {
     });
   }());
 
-  const proxyOpts = {
-    logLevel: 'warn',
-    target: `http://${canonicalHost(host)}:${port}`,
-    ws: ws,
-
-    onError(err, _req, _res) {
-      console.error(
-        __('Proxy: Error forwarding requests to blockchain/simulator'),
-        err.message
-      );
+  let proxy = httpProxy.createProxyServer({
+    target: {
+      host: canonicalHost(host),
+      port: port
     },
+    ws: ws,
+    createWsServerTransformStream: (_req, _proxyReq, _proxyRes) => {
+      const parser = new WsParser(0, true);
+      parser.on('frame', ({data: buffer}) => {
+        let object = parseJsonMaybe(buffer.toString());
+        if (object) {
+          object = modifyPayload(toModifyPayloads, object, accounts);
+          // track the modified response
+          trackResponse(object);
+          // send the modified response
+          WsWrapper.wrap(
+            {connection: dupl, masked: 0},
+            Buffer.from(JSON.stringify(object)),
+            () => {}
+          );
+        }
+      });
+      const dupl = new Duplex({
+        read(_size) {},
+        write(chunk, encoding, callback) {
+          parser.write(chunk);
+          callback();
+        }
+      });
+      return dupl;
+    }
+  });
 
-    onProxyReq(_proxyReq, req, _res) {
-      if (req.method === 'POST') {
-        // messages TO the target
-        Asm.connectTo(
-          pump(req, jsonParser())
-        ).on('done', ({current: object}) => {
-          trackRequest(object);
-        });
-      }
-    },
+  proxy.on('error', (err) => {
+    console.error(
+      __('Proxy: Error forwarding requests to blockchain/simulator'),
+      err.message
+    );
+  });
 
-    onProxyRes(proxyRes, req, _res) {
-      if (req.method === 'POST') {
-        // messages FROM the target
-        Asm.connectTo(
-          pump(proxyRes, jsonParser())
-        ).on('done', ({current: object}) => {
-          trackResponse(object);
-        });
+  proxy.on('proxyRes', (proxyRes, req, res) => {
+    modifyResponse(res, proxyRes, (body) => {
+      if (body) {
+        body = modifyPayload(toModifyPayloads, body, accounts);
+        trackResponse(body);
       }
-    }
-  };
+      return body;
+    });
+  });
 
-  if (ws) {
-    proxyOpts.onProxyReqWs = (_proxyReq, _req, socket, _options, _head) => {
+  const server = http.createServer((req, res) => {
+    if (req.method === 'POST') {
       // messages TO the target
-      pump(socket, new WsParser(0, false)).on('frame', ({data: buffer}) => {
-        const object = parseJsonMaybe(buffer.toString());
+      Asm.connectTo(
+        pump(req, jsonParser())
+      ).on('done', ({current: object}) => {
         trackRequest(object);
       });
-    };
+    }
+
+    if (!ws) {
+      proxy.web(req, res);
+    }
+  });
+
+  if (ws) {
+    server.on('upgrade', function (msg, socket, head) {
```

Comment:
Can we go with arrow function here too? :)

Reply:
Done.